### PR TITLE
numa_memory_spread: Fix numactl: mmap: Cannot allocate memory

### PR DESCRIPTION
**Pull request by liuyd96**
Even if virsh setmem return successfully, guest still need some time to
increase memory.

Signed-off-by: Liu Yiding <liuyd.fnst@fujitsu.com>

Before
```
# avocado run  --vt-type libvirt --vt-arch aarch64 --vt-machine-type arm64-mmio numa_memory_spread.default
JOB ID     : 83f557a3a0c59aa150f3c6c459f9007d94c60ca4
JOB LOG    : /root/avocado/job-results/job-2021-10-20T05.40-83f557a/job.log
 (1/1) type_specific.io-github-autotest-libvirt.numa_memory_spread.default: ERROR: Shell command failed: 'memhog -r1 8649555k'    (status: 1,    output: 'numactl: mmap: Cannot allocate memory\n') (73.80 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 74.51 s
```

After
```
# avocado run  --vt-type libvirt --vt-arch aarch64 --vt-machine-type arm64-mmio numa_memory_spread.default
JOB ID     : e3748056f67e42b33a69e0a363c7aedbf50783ba
JOB LOG    : /root/avocado/job-results/job-2021-10-20T05.44-e374805/job.log
 (1/1) type_specific.io-github-autotest-libvirt.numa_memory_spread.default: ERROR: Timeout expired while waiting for shell command to complete: 'memhog -r1 8649555k'    (output: '.................................................................................................................................................................. (180.41 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 181.13 s
```

### CODE REVIEW

**Comment by liuyd96** [comment]
@Yingshun . Please help me to review it. Thanks.

File: libvirt/tests/src/numa/numa_memory_spread.py [hunk 1]
```diff
@@ -106,6 +110,22 @@ def prepare_guest_for_test(vm_name, session, test, oversize, nodeset_string):
     if result.exit_status:
         test.fail("Something went wrong during the 'virsh setmem {}' "
                   "command.".format(oversize))
+
+    # Need some time to increase guest memory
+    end_time = time.time() + 300
```
Comment (Yingshun) [inline libvirt/tests/src/numa/numa_memory_spread.py:115]: How about calling utils_mis.wait_for here? It looks simpler I think.
for eg. 
```
def _check_mem(memory_to_eat):
            dommemstat_output = virsh.dommemstat(vm_name).stdout_text.strip()
            actual_mem = re.search("actual (\d*)", dommemstat_output).groups()[0]
            logging.debug("actual_mem is {}".format(actual_mem))
            return int(actual_mem) > int(memory_to_eat)
if not utils_misc.wait_for(lambda: _check_mem(memory_to_eat), 300, first=5):
    test.error("xxx")
```

Comment (liuyd96) [reply libvirt/tests/src/numa/numa_memory_spread.py]: Great ! It's quiet neat !

Comment (liuyd96) [reply libvirt/tests/src/numa/numa_memory_spread.py]: > How about calling utils_mis.wait_for here? It looks simpler I think. for eg.
> 
> ```
> def _check_mem(memory_to_eat):
>             dommemstat_output = virsh.dommemstat(vm_name).stdout_text.strip()
>             actual_mem = re.search("actual (\d*)", dommemstat_output).groups()[0]
>             logging.debug("actual_mem is {}".format(actual_mem))
>             return int(actual_mem) > int(memory_to_eat)
> if not utils_misc.wait_for(lambda: _check_mem(memory_to_eat), 300, first=5):
>     test.error("xxx")
> ```

Great ! It's quiet neat !

**Review by kamilvarga — APPROVED** [review]
LGTM.

**Review by Yingshun — APPROVED** [review]

**Pull request merged**